01_Data/src/layers/sequelize/repository/DeviceModel.ts

```diff
@@ -290,9 +290,9 @@ export class DeviceModelRepository extends SequelizeRepository<VariableAttribute
     return {
       where: {
         ...(queryParams.stationId ? { stationId: queryParams.stationId } : {}),
-        ...(queryParams.type !== null ? { type: queryParams.type } : {}),
```

Comment:
The original intent of these checks was to permit explicit searching for cases where the type or status was null. I.e., _undefined_ type (missing from query) would be interpreted as any type, but null type (present in query as null) would be interpreted as no type.

This functionality doesn't look like it works as written either. Is there a way we can fix it that provides the intended functionality?

Reply:
Thanks for the feedback, Thana! I updated the code. Here are some tests. 
1. filter type by null, only one attribute has null type returns:  <img width="1274" alt="Screenshot 2024-05-02 at 1 43 37 PM" src="https://github.com/citrineos/citrineos-core/assets/21100540/bdf2a0ee-deca-4183-9537-f2e7107ae1bf">
2. undefined type in the query then all attributes return <img width="1298" alt="Screenshot 2024-05-02 at 1 44 51 PM" src="https://github.com/citrineos/citrineos-core/assets/21100540/f5f9b8c0-f0e9-4817-a53b-13de49c6c6d4">
3. The message endpoint works  <img width="1154" alt="Screenshot 2024-05-02 at 1 55 10 PM" src="https://github.com/citrineos/citrineos-core/assets/21100540/54f2c7df-e13a-4514-8aa0-b7c7aa43b940">